Memory Allocation in Operating Systems in Java

shraiyya opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.
You can learn more about contributing to NeoAlgo here: https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md

Happy Contributing!

-->

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

PR corresponding to issue #1984 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Please add the DWOC tag and level of the code.

shraiyya committed:
Merge pull request #5 from TesseractCoding/master

Fork updation

shraiyya committed:
Create alloc.java

shraiyya committed:
Update README.md

shraiyya committed:
Merge pull request #6 from TesseractCoding/master

Updation of my fork

HarshCasper commented on Java/Operating-System/alloc.java:
```diff
@@ -0,0 +1,172 @@
+Title: Memory Allocation in Operating Systems in Java
+
+Description:
+(a) First-Fit: Allocate the first hole that is big enough. Searching can start either at the beginning of the set of holes or where the previous first search ended. We can stop searching as soon as we find a free hole that is large enough.
```
Break down the line into shorter, more consumable lines.

HarshCasper commented on Java/Operating-System/alloc.java:
```diff
@@ -0,0 +1,172 @@
+import java.util.Scanner;
+public class alloc {
+
+  public static void main(String[] args) {
```
Keep your main() function only for the Driver Code. Break down your entire code into modular functions and then call them from the main() function

HarshCasper commented on Java/Operating-System/alloc.java:
```diff
@@ -0,0 +1,172 @@
+
+  }
+
+}
```
Add sample I/O in a Multi-Line Comment

shraiyya committed:
Added sample I/O in a Multi-Line Comment

shraiyya committed:
Made other changes

Brokedown your entire code into modular functions
called them from the main() function
Brokedown the lines into shorter sentences

shraiyya commented:
Please add the level of the code.
